### PR TITLE
Fix links in tabs

### LINKED ISSUE #1227: 404 for latest execution info from `run` table
From main page of project we get: https://saveourtool.com/#/CQFN.org/Diktat/history/execution/212 
But from `Run` tab we get: https://saveourtool.com/#/project/CQFN.org/Diktat/run/history/execution/212 which leads to 404

### PR DESCRIPTION
**Pull request by Arrgentum**
## Whats added
- Fix links inside tab in any views
- Correct gradle properties o run backend on linux
- Run diktat fix, deleted the commented code
- Create class PathsForTabs and redone using it

## Issus
This pull request close #1227

### CODE REVIEW

File: save-backend/src/main/resources/application-dev.properties [hunk 1]
```diff
@@ -5,4 +5,5 @@ datasource.dev.url=jdbc:mysql://localhost:3306/save_cloud
 spring.datasource.username=root
 spring.datasource.password=123
 backend.preprocessorUrl=http://localhost:5200
-backend.orchestratorUrl=http://localhost:5100
\ No newline at end of file
+backend.orchestratorUrl=http://localhost:5100
```
Comment (akuleshov7) [inline save-backend/src/main/resources/application-dev.properties:8]: Why changed?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ContestView.kt [hunk 2]
```diff
@@ -103,6 +107,9 @@ class ContestView : AbstractView<ContestViewProps, ContestViewState>(false) {
 
     override fun componentDidMount() {
         super.componentDidMount()
+        scope.launch {
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ContestView.kt:110]: Why do you need to launch a coroutine here?

Comment (akuleshov7) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ContestView.kt:110]: You are not doing any request

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt [hunk 3]
```diff
@@ -268,17 +274,18 @@ class OrganizationView : AbstractView<OrganizationProps, OrganizationViewState>(
...
-            buttonBuilder("Ok") {
-                when (state.confirmationType) {
-                    ConfirmationType.DELETE_CONFIRM -> deleteOrganizationBuilder()
-                    else -> throw IllegalStateException("Not implemented yet")
+        withNavigate {navigateContext ->
+            displayModal(state.isConfirmWindowOpen, state.confirmLabel, state.confirmMessage, smallTransparentModalStyle, { setState { isConfirmWindowOpen = false } }) {
+                buttonBuilder("Ok") {
+                    when (state.confirmationType) {
+                        ConfirmationType.DELETE_CONFIRM -> navigateContext.deleteOrganizationBuilder()
+                        else -> throw IllegalStateException("Not implemented yet")
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt:282]: What is this exception for? If you want to throw not implemented - use TODO

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt [hunk 4]
```diff
@@ -717,7 +724,7 @@ class OrganizationView : AbstractView<OrganizationProps, OrganizationViewState>(
         }
     }
 
-    private fun deleteOrganizationBuilder() {
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt:720]: Why this method removed?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ProjectView.kt [hunk 5]
```diff
@@ -622,22 +630,22 @@ class ProjectView : AbstractView<ProjectExecutionRouteProps, ProjectViewState>(f
...
+                                    navigateContext.generateLinksWithSuffix<ProjectMenuBar>(state.paths.pathDefaultTab, "history/execution/${state.latestExecutionId}")
+                                }
                             }
                         }
                     }
                     div {
                         className = ClassName("ml-3 align-items-left")
                         fontAwesomeIcon(icon = faCalendarAlt)
                         a {
-                            href = "#/${state.project.organization.name}/${state.project.name}/history"
```
Comment (akuleshov7) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ProjectView.kt:640]: Why slash in the beginning of the value is removed?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt [hunk 6]
```diff
@@ -21,6 +21,16 @@ external interface HasSelectedMenu<T : Enum<T>> : State {
     var selectedMenu: T
 }
 
+/**
+ * The class is needed to store the paths of different tabs in different Views
+ * @property pathDefaultTab (must not contain a "#" for the correct execution of [navigate] in the [generateLinksWithSuffix] function)
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt:26]:
```suggestion
 * @property pathDefaultTab (must not start with a "#" for the correct execution of [navigate] in the [generateLinksWithSuffix] function)
```

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt [hunk 7]
```diff
@@ -46,23 +56,29 @@ fun <T : Enum<T>, S : HasSelectedMenu<T>> AbstractView<*, S>.urlAnalysis(menu: T
     }
 }
 
+/**
+ * @param pathDefaultTab
+ * @param suffix
+ */
+fun <T : Enum<T>>NavigateFunctionContext.generateLinksWithSuffix(pathDefaultTab: String, suffix: String) {
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt:63]:
```suggestion
fun <T : Enum<T>>NavigateFunctionContext.navigateToLinkWithSuffix(pathDefaultTab: String, suffix: String) {
```
This doesn't generate anything, this performs navigation

Comment (Arrgentum) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt]: Done

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt (hunk 6 above)
Comment (Arrgentum) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt]: Done

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ProjectView.kt (hunk 5 above)
Comment (Arrgentum) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ProjectView.kt:640]: Sorry, added this

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt (hunk 3 above)
Comment (Arrgentum) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt]: Done

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ContestView.kt (hunk 2 above)
Comment (Arrgentum) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ContestView.kt]: Sorry, deleted this

File: save-backend/src/main/resources/application-dev.properties (hunk 1 above)
Comment (Arrgentum) [reply save-backend/src/main/resources/application-dev.properties:8]: this particular line has not changed.
added a line (subsequent) to run the backend under the linux system

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt (hunk 4 above)
Comment (Arrgentum) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt]: I wanted to make this method an extension function

**Review by petertrr — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt [hunk 8]
```diff
@@ -46,23 +56,29 @@ fun <T : Enum<T>, S : HasSelectedMenu<T>> AbstractView<*, S>.urlAnalysis(menu: T
     }
 }
 
+/**
+ * @param pathDefaultTab
+ * @param suffix
+ */
+fun <T : Enum<T>>NavigateFunctionContext.navigateToLinkWithSuffix(pathDefaultTab: String, suffix: String) {
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt:63]: where T is used?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt [hunk 9]
```diff
@@ -21,6 +21,16 @@ external interface HasSelectedMenu<T : Enum<T>> : State {
     var selectedMenu: T
 }
 
+/**
+ * The class is needed to store the paths of different tabs in different Views
+ * @property pathDefaultTab (must not start with a "#" for the correct execution of [navigate] in the [navigateToLinkWithSuffix] function)
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt:26]: > must not start with a "#"
maybe add init section with validating ?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt [hunk 10]
```diff
@@ -21,6 +21,16 @@ external interface HasSelectedMenu<T : Enum<T>> : State {
...
+ * @property extendedViewPath
+ */
+data class PathsForTabs(
+    val pathDefaultTab: String,
+    val extendedViewPath: String
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt:31]: maybe add a custom constructor with
```kotlin
extendedViewPath = "#$pathDefaultTab"
```

**Review by nulls — CHANGES_REQUESTED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt (hunk 10 above)
Comment (Arrgentum) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt:31]: No, these elements are not so interconnected,
by pathDefaultTab I meant the path for the default tab, and extended View Path is the prefix of the path for the rest of the tab (there is usually some word added at the beginning, for example, Project or Organization).

But I realized that it is not clear from the name, now I will add KDoc

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ProjectView.kt (hunk 5 above)
Comment (Arrgentum) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/ProjectView.kt:640]: ```pathDefaultTab``` start with ```\```

Need to store it ```pathDefaultTab```, for a good transition to other pages, using ```navigate``` (the url should not start with ```#```, but with ```/```). Therefore, it is used everywhere, both for navigating by ```navigate```, and for the usual change of href when switching between tabs(change ```window.location.href```)

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt (hunk 8 above)
Comment (Arrgentum) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt]: Deleted

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt (hunk 9 above)
Comment (Arrgentum) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/ParseUrlInTab.kt]: Done, thanks

**Review by nulls — APPROVED** [review]

**Pull request merged**